### PR TITLE
Fix Barbeque::SnsSubscriptionsController#fetch_sns_topic_arns

### PR DESCRIPTION
**Pull request by itkq**
Aws::SNS::Client#list_topics returns up to 100 topics. This
change allows to fetch all topics when there are more than 100
topics.

### CODE REVIEW

File: app/controllers/barbeque/sns_subscriptions_controller.rb [hunk 1]
```diff
@@ -44,6 +44,16 @@ def destroy
   private
 
   def fetch_sns_topic_arns
-    Barbeque::SNSSubscriptionService.sns_client.list_topics.topics.map(&:topic_arn)
+    topic_arns = []
+    next_token = nil
+
+    loop do
+      list_topics = Barbeque::SNSSubscriptionService.sns_client.list_topics(next_token: next_token)
```
Comment (riseshia) [inline app/controllers/barbeque/sns_subscriptions_controller.rb]: FYI https://github.com/aws/aws-sdk-ruby#paging-responses

> To make it easy to get the next page of results, every AWS response object is enumerable:

**Comment by riseshia** [comment]
cc @cookpad/dev-infra

File: app/controllers/barbeque/sns_subscriptions_controller.rb [hunk 2]
```diff
@@ -44,6 +44,22 @@ def destroy
   private
 
   def fetch_sns_topic_arns
-    Barbeque::SNSSubscriptionService.sns_client.list_topics.topics.map(&:topic_arn)
+    each_sns_topic_arns.to_a
+  end
+
+  def each_sns_topic_arns
+    unless block_given?
```
Comment (riseshia) [inline app/controllers/barbeque/sns_subscriptions_controller.rb]: is there any reason you want to support passing block?

File: app/controllers/barbeque/sns_subscriptions_controller.rb [hunk 3]
```diff
@@ -44,6 +44,6 @@ def destroy
   private
 
   def fetch_sns_topic_arns
-    Barbeque::SNSSubscriptionService.sns_client.list_topics.topics.map(&:topic_arn)
+    Barbeque::SNSSubscriptionService.sns_client.list_topics.flat_map{ |resp| resp.topics.map(&:topic_arn) }
```
Comment (ganmacs) [inline app/controllers/barbeque/sns_subscriptions_controller.rb]: This is enought, isn't it.
`Barbeque::SNSSubscriptionService.sns_client.list_topics.flat_map(&:topics).map(&:topic_arn) }`

Comment (itkq) [reply app/controllers/barbeque/sns_subscriptions_controller.rb]: Fixed. b28f812

**Comment by riseshia** [comment]
LGTM

**Pull request merged**